imagetest/test_suites/image_validation/license_test.go

```diff
@@ -0,0 +1,190 @@
+	`are in the public domain`,
+	`is (available|distributed) under the terms of the GNU.*Public License`,
+	`(libudev|libgudev|udev) is licensed under the GNU (L|)GPL`,
+	`The Linux Console Tools are covered by the GPL`,
+	`Some portions of os-prober`,
+	`Netcat and the associated package is a product of Avian Research, and is freely available in full source form with no restrictions save an obligation to give credit where due.`,
+	`Permission is hereby granted, without written agreement and without licence or royalty fees, to use, copy, modify, and distribute this software`,
+	`Open Market permits you to use, copy, modify, distribute, and license this Software and the Documentation for any purpose, provided that existing copyrightPathGlob notices are retained in all copies and that this notice is included verbatim in any distributions. No written agreement, license, or royalty fee is required for any of the authorized uses.`,
+	`This software is made available under the terms of *either* of the licenses found in LICENSE.APACHE or LICENSE.BSD. Contributions to cryptography are made under the terms of *both* these licenses.`,
+}
+
+const (
+	copyrightPathGlob = "/usr/share/doc/*/copyrightPathGlob"
+	licensePathGlob   = "/usr/share/doc/*/LICENSE"
+	licenseNameRegex  = `(?i)((?:(?:License|Copyright)\s*:\s*%[1]s)|(?:(?:covered )*under (?:the )?%[1]s)|(?:under (?:the terms of )*the %[1]s))`
+)
+
+func isValidLicenseName(licenseCheck string) bool {
+	for _, name := range licenseNames {
+		// (?i) case insensitive
+		var regexString = fmt.Sprintf(licenseNameRegex, name)
+		re := regexp.MustCompile(regexString)
+
+		if re.MatchString(licenseCheck) {
+			return true
+		}
+	}
+	return false
+}
+
+func isValidLicenseText(licenseCheck string) bool {
+	for _, licenseText := range licenses {
+		// (?i) case insensitive
+		re := regexp.MustCompile(`(?i)` + licenseText)
+
+		if re.MatchString(licenseCheck) {
+			return true
+		}
+	}
+	return false
+}
+
+func TestArePackagesLegal(t *testing.T) {
+	var filenames []string
+	if utils.IsTargetLinux(utils.RedHat) || utils.IsTargetLinux(utils.SUSE) {
+		filenames, _ = filepath.Glob(licensePathGlob)
+	} else if utils.IsTargetLinux(utils.Debian) {
+		filenames, _ = filepath.Glob(copyrightPathGlob)
+	} else {
+		t.Skip("OS not supported")
+	}
+	for _, filename := range filenames {
+		if !isPackageLegal(filename) {
+			t.Fatalf("Found illegal package: %v", filename)
+		}
+	}
+}
+
+func isPackageLegal(filepath string) bool {
+	bytes, err := ioutil.ReadFile(filepath)
+	if err != nil {
+		log.Printf("error read file")
+	}
+
+	var licenseCheck string = string(bytes)
+
+	// Remove comment
+	re := regexp.MustCompile(`(\*|#).`)
+	re.ReplaceAllString(licenseCheck, "")
+
+	// Replace all whitespace with one space
+	whitespaceRegex := regexp.MustCompile(`\s+`)
+	licenseCheck = whitespaceRegex.ReplaceAllString(licenseCheck, " ")
+	if !isValidLicenseName(licenseCheck) && isValidLicenseText(licenseCheck) {
```

Comment:
i don't think this logic is correct. one has a ! and one doesn't, and you're using &&
it reads like you would want to say if !.. || !.. meaning if either one fail, short circuiting on the first failure

Reply:
lost `!` for the second condition. tranlate from `!( ... || ...)` to `!... && !...`

Reply:
i see, a package has only failed if both have returned false
maybe invert the logic and say if isValidLicenseName || isValidLicenseText { return true }
this way it can stop as soon as it reaches success, and easier to read